sanic/app.py

```diff
@@ -213,6 +217,9 @@ def __init__(
         self.blueprints: Dict[str, Blueprint] = {}
         self.configure_logging: bool = configure_logging
         self.ctx: Any = ctx or SimpleNamespace()
+        self.directory_handler: DirectoryHandler = (
+            directory_handler or DirectoryHandler(self.debug)
+        )
```

Comment:
Suggest leave out this option entirely, keep it hard-coded at least for now, such that easier to change later.

Reply:
As mentioned on Discord, it would make a lot more sense to me to instantiate for each `app.static()` also handling the autoindex/index args there directly, rather than at runtime when a request is being handled.

Reply:
That just feels very inconsistent.

Reply:
Inconsistent against what?

Reply:
Our existing use of "Handler" objects.

Reply:
Okay... I wasn't aware of other than ErrorHandler and still not quite sure what the pattern is. But I am going to mark this resolved, in hopes for the whole class to eliminated as discussed in another thread, and to keep this discussion in that thread.